core/src/main/java/io/ultrabrew/metrics/reporters/TimeWindowReporter.java

```diff
@@ -200,16 +203,39 @@ public B addHistogram(final String metricId, final DistributionBucket bucket, fi
       return (B) this;
     }
 
-      /**
-       * Add histograms to a specific metric
-       *
-       * @param metricId identifier of the metric
-       * @param bucket distribution bucket
-       */
+    /**
+     * Add histograms to a specific metric
+     *
+     * @param metricId identifier of the metric
+     * @param bucket distribution bucket
+     * @param maxCardinality maximum cardinality of data in the histogram
+     */
+    public B addHistogram(final String metricId, final DoubleValuedDistributionBucket bucket, final int maxCardinality) {
```

Comment:
Is it worth making the API a bit cleaner and having a `DistributionBucket` interface then have `DoubleValuedDistributionBucket` implement it?

Reply:
That might break backward compatibility.

Reply:
I have added a common interface. Please take a look